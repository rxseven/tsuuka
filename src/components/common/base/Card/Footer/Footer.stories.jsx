import React from 'react';
import { storiesOf } from '@storybook/react';

import Footer from './index';

const title = 'Components/Common/Base/Card/Footer';
const component = <Footer>Card Footer</Footer>;

storiesOf(title, module).add('default', () => component);
