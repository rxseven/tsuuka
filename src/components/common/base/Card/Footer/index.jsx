import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Footer({ children }) {
  return <div className="card-footer text-muted">{children}</div>;
}

Footer.propTypes = propTypes;

export default Footer;
