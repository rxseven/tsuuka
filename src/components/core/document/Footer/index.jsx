import React from 'react';
import styled, { css } from 'styled-components';

import Grid from 'components/common/base/Grid';
import Hyperlink from 'components/common/base/Hyperlink';
import Nav from 'components/common/base/Nav';
import PATHS from 'constants/routes/paths';

const linkStyles = css`
  color: #999;
  transition: color 0.2s ease-in-out;

  :hover {
    color: #fff;
  }
`;

const Frame = styled.footer`
  background-color: #343a40;
  color: #999;
  flex-shrink: 0;
  padding: 1.5rem 0;
  z-index: 2;

  @media (min-width: ${({ theme }) => theme.breakpoint.sm}) {
    z-index: 0;
  }
`;

const Info = styled.div`
  text-align: center;

  @media (min-width: ${({ theme }) => theme.breakpoint.sm}) {
    text-align: left;
  }
`;

const Navigation = styled.ul`
  list-style: none;
  margin: 0 0 1.25rem 0;
  padding: 0;
  text-align: center;

  @media (min-width: ${({ theme }) => theme.breakpoint.sm}) {
    text-align: left;
  }
`;

const NavItem = styled.li`
  display: inline;
  font-size: 0.75rem;

  & + & {
    margin-left: 1rem;
  }
`;

const NavLink = styled(Nav.Link).attrs({ light: true })`
  ${linkStyles}
`;

const Link = styled(Hyperlink)`
  :hover {
    text-decoration: none;
  }

  ${linkStyles}
`;

const Text = styled.p`
  font-size: 0.75rem;
  margin-bottom: 0.25rem;

  :last-child {
    margin-bottom: 0;
  }
`;

function Footer() {
  return (
    <Frame>
      <Grid.Container>
        <Grid.Row>
          <Grid.Column>
            <Navigation>
              <NavItem>
                <NavLink exact to={PATHS.root}>
                  Currency Converter
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink to={PATHS.about}>About</NavLink>
              </NavItem>
              <NavItem>
                <NavLink to={PATHS.terms}>Terms</NavLink>
              </NavItem>
              <NavItem>
                <Link href="https://tsuuka.netlify.com">Storybook</Link>
              </NavItem>
              <NavItem>
                <Link href="https://github.com/rxseven/tsuuka">GitHub</Link>
              </NavItem>
            </Navigation>
            <Info>
              <Text>Designed &amp; built with all the love in React.</Text>
              <Text>Copyright Â© 2019 Theerawat Pongsupawat.</Text>
            </Info>
          </Grid.Column>
        </Grid.Row>
      </Grid.Container>
    </Frame>
  );
}

export default Footer;
