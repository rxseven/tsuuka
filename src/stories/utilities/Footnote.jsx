import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
