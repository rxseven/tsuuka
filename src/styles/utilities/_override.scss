// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
