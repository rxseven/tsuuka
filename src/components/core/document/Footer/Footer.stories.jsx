import React from 'react';
import { storiesOf } from '@storybook/react';

import Footer from './index';

const title = 'Components/Core/Document/Footer';
const component = <Footer />;

storiesOf(title, module).add('default', () => component);
