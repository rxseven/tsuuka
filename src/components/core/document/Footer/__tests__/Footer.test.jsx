import { factory } from 'tests/utilities';
import Footer from '../index';

// Arrange
const source = {};

// Setup
function setup(props) {
  return factory(Footer, source, props);
}

// Test suites
describe('<Footer />', () => {
  it('should render without crashing', () => {
    setup();
  });
});
