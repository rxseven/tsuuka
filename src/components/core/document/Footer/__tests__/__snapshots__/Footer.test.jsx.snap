// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Snapshot tests should render correctly 1`] = `
<footer
  className="sc-bwzfXH cNWnVm"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <ul
          className="sc-bxivhb hpIYpU"
        >
          <li
            className="sc-ifAKCX kQYkju"
          >
            <a
              aria-current="page"
              className="sc-EHOje hlRBmM sc-bdVaJa hYPwXj active"
              data-light={true}
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Currency Converter
            </a>
          </li>
          <li
            className="sc-ifAKCX kQYkju"
          >
            <a
              aria-current={null}
              className="sc-EHOje hlRBmM sc-bdVaJa hYPwXj"
              data-light={true}
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="sc-ifAKCX kQYkju"
          >
            <a
              aria-current={null}
              className="sc-EHOje hlRBmM sc-bdVaJa hYPwXj"
              data-light={true}
              href="/terms"
              onClick={[Function]}
            >
              Terms
            </a>
          </li>
          <li
            className="sc-ifAKCX kQYkju"
          >
            <a
              className="sc-bZQynM htyeqS"
              href="https://tsuuka.netlify.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Storybook
            </a>
          </li>
          <li
            className="sc-ifAKCX kQYkju"
          >
            <a
              className="sc-bZQynM htyeqS"
              href="https://github.com/rxseven/tsuuka"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </a>
          </li>
        </ul>
        <div
          className="sc-htpNat ccJztr"
        >
          <p
            className="sc-gzVnrw cacXWT"
          >
            Designed & built with all the love in React.
          </p>
          <p
            className="sc-gzVnrw cacXWT"
          >
            Copyright Â© 2019 Theerawat Pongsupawat.
          </p>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
